src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py

```diff
@@ -129,13 +133,16 @@ def get_cluster_name(self) -> str:
         # this parameter does not need validation
         return cluster_name
 
-    def get_nodepool_name(self) -> str:
-        """Obtain the value of nodepool_name.
+    def _get_nodepool_name(self, enable_validation: bool = False) -> str:
+        """Internal function to obtain the value of nodepool_name.
 
         Note: SDK performs the following validation {'required': True, 'pattern': r'^[a-z][a-z0-9]{0,11}$'}.
 
         This is a required parameter and its value should be provided by user explicitly.
 
+        This function supports the option of enable_validation. When enabled, it will check if the given nodepool name
```

Comment:
please add some comment where we should set the enable_validation to false.

Reply:
sure

Reply:
Updated in commit [#1c010d9](https://github.com/Azure/azure-cli/pull/21543/commits/1c010d946e9d5172dcba35e94d08ddcdc8e34887).